src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -415,4 +424,46 @@ void processRebalancingTest() {
       }
     }
   }
+
+  /* Useful to report a record back to DLQ if error tolerance is specified */
+  private KafkaRecordErrorReporter createKafkaRecordErrorReporter() {
+    KafkaRecordErrorReporter result = noOpKafkaRecordErrorReporter();
+    if (context != null) {
+      try {
+        ErrantRecordReporter errantRecordReporter = context.errantRecordReporter();
+        if (errantRecordReporter != null) {
+          result =
+              (record, error) -> {
+                try {
+                  // Blocking this until record is delivered to DLQ
+                  errantRecordReporter.report(record, error).get();
+                } catch (InterruptedException | ExecutionException e) {
+                  final String errMsg = "ERROR reporting records to ErrantRecordReporter";
```

Comment:
Nit, inline `errMsg` below?

Reply:
Wanted to use it for logging as well as in connect exception.